controllers/keda/hpa.go

```diff
@@ -102,6 +102,19 @@ func (r *ScaledObjectReconciler) newHPAForScaledObject(ctx context.Context, logg
 	minReplicas := scaledObject.GetHPAMinReplicas()
 	maxReplicas := scaledObject.GetHPAMaxReplicas()
 
+	if *minReplicas == 0 {
+		minReplicas = scaledObject.GetDefaultHPAMinReplicas()
+	}
+
+	if maxReplicas == 0 {
+		maxReplicas = scaledObject.GetDefaultHPAMaxReplicas()
+	}
+
+	if *minReplicas < 0 || maxReplicas < 0 {
```

Comment:
nit: no need to check `max` for less than `0`, it has to be bigger than `min` and `min` is checked for larger than `0` so transitively `max` is already larger

Reply:
> nit: no need to check `max` for less than `0`, it has to be bigger than `min` and `min` is checked for larger than `0` so transitively `max` is already larger

make sense